activejob/lib/active_job/arguments.rb

```diff
@@ -101,7 +109,7 @@ def deserialize_argument(argument)
         case argument
         when String
           argument
-        when *PERMITTED_TYPES
+        when *PERMITTED_TYPES, BigDecimal # BigDecimal may have been legacy serialized
```

Comment:
This ensures the deserialization behaviour matches what it was previously, since `BigDecimal` used to be in `PERMITTED_TYPES`.

Without this, and `BigDecimal` arguments serialized without `BigDecimalSerializer` would  lead to the `ArgumentError` `else` branch.

Reply:
We shouldn't forget to remove it in 7.2, maybe make it its own `when` with a comment saying to remove in 7.2